addons/website/models/ir_http.py

```diff
@@ -62,6 +64,57 @@ def get_request_website():
 class Http(models.AbstractModel):
     _inherit = 'ir.http'
 
+    @classmethod
+    def routing_map(cls, key=None):
+        key = key or (request and request.website_routing)
+        return super(Http, cls).routing_map(key=key)
+
+    @classmethod
+    def clear_caches(cls):
+        super(Http, cls)._clear_routing_map()
+        return super(Http, cls).clear_caches()
+
+    @classmethod
+    def _slug_matching(cls, adapter, endpoint, **kw):
+        for arg in kw:
+            if isinstance(kw[arg], models.BaseModel):
+                kw[arg] = kw[arg].with_user(request.uid)
+        qs = request.httprequest.query_string.decode('utf-8')
+        return adapter.build(endpoint, kw) + (qs and '?%s' % qs or '')
+
+    @classmethod
+    def _match(cls, path_info, key=None):
+        key = key or (request and request.website_routing)
+        return super(Http, cls)._match(path_info, key=key)
+
+    @classmethod
+    def _generate_routing_rules(cls, modules, converters):
+        website_id = request.website_routing
+        logger.debug("_generate_routing_rules for website: %s", website_id)
+        domain = ['|', ('redirect_type', '=', 'rewrite'), ('redirect_type', '=', 'not_found'), '|', ('website_id', '=', False), ('website_id', '=', website_id)]
+
+        rewrites = dict([(x.url_from, x) for x in request.env['website.rewrite'].sudo().search(domain)])
+
+        for u, e, r in super(Http, cls)._generate_routing_rules(modules, converters):
+            r = dict(r)
+            if u in rewrites:
+                rewrite = rewrites[u]
+                url_to = rewrite.url_to
+                if rewrite.redirect_type == 'rewrite' and u != url_to:
```

Comment:
When `u == url_to`, the url is ignored and act as a 404. It should be yield (once)

Reply:
:+1: